pkg/controller/dynamodb/table/hooks.go

```diff
@@ -177,18 +174,18 @@ func lateInitialize(in *svcapitypes.TableParameters, t *svcsdk.DescribeTableOutp
 			WriteCapacityUnits: t.Table.ProvisionedThroughput.WriteCapacityUnits,
 		}
 	}
-	if in.SSESpecification == nil && t.Table.SSEDescription != nil {
-		in.SSESpecification = &svcapitypes.SSESpecification{
-			SSEType: t.Table.SSEDescription.SSEType,
+	if t.Table.SSEDescription != nil {
+		if in.SSESpecification == nil {
+			in.SSESpecification = &svcapitypes.SSESpecification{}
+		}
+		if in.SSESpecification.Enabled == nil && t.Table.SSEDescription.Status != nil {
+			in.SSESpecification.Enabled = aws.Bool(*t.Table.SSEDescription.Status == string(svcapitypes.SSEStatus_ENABLED))
 		}
-	}
-	if in.SSESpecification != nil {
 		if in.SSESpecification.KMSMasterKeyID == nil && t.Table.SSEDescription.KMSMasterKeyArn != nil {
```

Comment:
I'm pretty sure this refactoring fixes an NPE we had in master that no one hit (at least until I was testing this out). It was possible that:

* `in.SSESpecification` was not nil.
* `in.SSESpecification.KMSMasterKeyID` was nil.
* `t.Table.SSEDescription` was nil

This would happen if (for example) someone was trying to enable SSE by just setting `enabled = true` without including an ARN.